lib/Doctrine/ORM/EntityManager.php

```diff
@@ -613,7 +613,7 @@ public function newHydrator(string|int $hydrationMode): AbstractHydrator
                 }
         }
 
-        throw InvalidHydrationMode::fromMode((string) $hydrationMode);
```

Comment:
I think, the string cast should stay.

Reply:
I removed it because Psalm right infers that it is redundant according to this:

https://github.com/doctrine/orm/blob/3dc5581294ccc05e0150c8b6708ccefbea88849e/lib/Doctrine/ORM/EntityManagerInterface.php#L298

Reply:
I think, we've been here before. That annotation prevents nobody from passing an integer that is not in the `Query::HYDRATE_*` range. Psalm is wrong complaining about this kind of "redundancy". "Fixing" this kind of error would basically mean removing input validation from all methods.

Note that this very piece of code is reached specifically for invalid input. So we have to assume that the called passed something that does not match the doc block.

Reply:
I've restored the string cast.

Reply:
You did not, apparently. 🤔

Reply:
Restored for real this time!